### PR TITLE
docs: remove running cs locally from development set up guide

### PR DESCRIPTION
**Pull request by JameelB**
### What this PR does
Removes `CS local Development Setup` section from the development set up guide. This should be replaced with how to connect and interact with the deployed Cluster Service for creating ARO HCP cluster and node pools.

All of the documentation for the CS local dev set up has now been moved to the CS repo: https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/merge_requests/8862

Jira: N/A
Link to demo recording: N/A

### CODE REVIEW

File: dev-infrastructure/docs/development-setup.md
Comment (ahitacat) [inline dev-infrastructure/docs/development-setup.md:385]: 'resource_name', 'resource_group_name' and 'managed_resources_group_name' can be a bit confusing, we have some explanation around this in the CS repo. Should we point there or copy the same info here?

**Review by ahitacat — COMMENTED** [review]
Overall this LGTM, I just have one small comment.

Thanks @JameelB to take care of this.

File: dev-infrastructure/docs/development-setup.md
Comment (patilsuraj767) [inline dev-infrastructure/docs/development-setup.md:326]:
```suggestion
    KUBECONFIG=$(make infra.svc.aks.kubeconfigfile) kubectl port-forward svc/clusters-service 8000:8000 -n cluster-service
```

File: dev-infrastructure/docs/development-setup.md
Comment (patilsuraj767) [inline dev-infrastructure/docs/development-setup.md:337]:
```suggestion
  - Create a resource group for the virtual network that will host your ARO HCP cluster(Data plane nodes).
```
Worker nodes are created under `managed_resource_group_name`, they are not created in this resource group. This resource group only holds Vnet and NSG.

Comment (JameelB) [reply dev-infrastructure/docs/development-setup.md:337]: that's correct, but this is talking about the usage of the virtual network not the resource group itself. I'll reword it a bit to make this clearer.

File: dev-infrastructure/docs/development-setup.md
Comment (JameelB) [reply dev-infrastructure/docs/development-setup.md:385]: We can't point to the CS repo as external users will not have access to it. However, all of this information should be available in our API docs which is available publicly in api.openshift.com, see the note at the start of this section. That should contain details to clarify these properties.

File: dev-infrastructure/docs/development-setup.md
Comment (patilsuraj767) [reply dev-infrastructure/docs/development-setup.md]: Thanks @JameelB LGTM 👍

File: dev-infrastructure/docs/development-setup.md
Comment (bennerv) [inline dev-infrastructure/docs/development-setup.md:400]: Remove trailing comma
```suggestion
        "np_provisioner_deprovision_enabled": "true"
```

Comment (JameelB) [reply dev-infrastructure/docs/development-setup.md:400]: good spot, thanks!

**Comment by JameelB** [comment]
@bennerv I see you've enabled auto merge here, I think it needs approval before it can be merged? 🤔

**Review by bennerv — APPROVED** [review]

**Pull request merged**
